src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetAssetResolver.cs

```diff
@@ -46,7 +46,8 @@ public ContentItemGroup GetCompileItems(NuGetFramework framework)
             // compile falls back to runtime pattern in the case of lib with no ref, this won't fallback
             // to a runtime asset since we have no RID passed in.
             return _sourceItems.FindBestItemGroup(managedCriteria,
-                _conventions.Patterns.CompileAssemblies,
+                // NOTE: This is a guess! https://github.com/NuGet/NuGet.Client/commit/69c37fdbd66256a3c8fdb3080ef51b1ce840fdf4
```

Comment:
This is calling out that I don't fully understand the change, with a pointer to the API break that made me have to do something. Here, `CompileAssemblies` was changed to `CompileRefAssemblies`, and `CompileLibAssemblies` was added, but I don't know if `CompileLibAssemblies` now needs to be handled somehow. Adding both caused test errors, so I ended up adding only `CompileRefAssemblies`.

I'm still hoping that https://github.com/NuGet/Home/issues/5139 will be fixed rather than have to make this change. I definitely don't plan on merging the note in.